mdtraj/core/topology.py

```diff
@@ -592,9 +597,19 @@ def from_dataframe(
             if col not in atoms.columns:
                 raise ValueError("dataframe must have column %s" % col)
 
+        if "formal_charge" not in atoms.columns:
+            atoms["formal_charge"] = None
+
+
         if "segmentID" not in atoms.columns:
             atoms["segmentID"] = ""
 
+        # Ensure formal_charge is of type Int64 (nullable integer)
+        try:
+            atoms = atoms.astype({"formal_charge": "Int64"})
+        except ValueError as e:
+            raise ValueError(f"Error converting formal_charge to Int64: {e}")
```

Comment:
Could this say something else about checking to make sure your charges are integer values?

Reply:
Sure, thanks for spotting that